frontend/packages/knative-plugin/src/components/pub-sub/form-fields/PubSubSubscriber.tsx

```diff
@@ -44,6 +45,14 @@ const PubSubSubscriber: React.FC = () => {
     setResourceAlert(_.isEmpty(resourceList));
   };
 
+  // filter out resource which are owned by other resource
+  const resourceFilter = (resource: K8sResourceKind) => {
+    const {
+      metadata: { ownerReferences },
+    } = resource;
+    return !ownerReferences?.length;
+  };
+
```

Comment:
Can we move this to util as it's duplicate code at 2 places?

Reply:
have updated it, and it makes sense to have `resourceFilter` in the respective file for better readability as this is passed to `ResourceDropdownField`